pkg/apis/core/validation/shootextensionstatus.go

```diff
@@ -0,0 +1,71 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"github.com/gardener/gardener/pkg/apis/core"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateShootExtensionStatus validates a ShootExtensionStatus object
+func ValidateShootExtensionStatus(ShootExtensionStatus *core.ShootExtensionStatus) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&ShootExtensionStatus.ObjectMeta, true, apivalidation.NameIsDNSLabel, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateShootExtensionStatusSpec(&ShootExtensionStatus.Spec, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+// ValidateShootExtensionStatusUpdate validates an update to a ShootExtensionStatus object
+func ValidateShootExtensionStatusUpdate(newShootExtensionStatus, oldShootExtensionStatus *core.ShootExtensionStatus) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newShootExtensionStatus.ObjectMeta, &oldShootExtensionStatus.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateShootExtensionStatusSpecUpdate(&newShootExtensionStatus.Spec, &oldShootExtensionStatus.Spec)...)
+	allErrs = append(allErrs, ValidateShootExtensionStatus(newShootExtensionStatus)...)
+
+	return allErrs
+}
+
+// ValidateShootExtensionStatusSpec validates the spec field of a ShootExtensionStatus object.
+func ValidateShootExtensionStatusSpec(ShootExtensionStatusSpec *core.ShootExtensionStatusSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i, data := range ShootExtensionStatusSpec.ExtensionStatus {
+		idxPath := fldPath.Child("extensionStatus").Index(i)
+
+		if len(data.Kind) == 0 {
+			kindPath := idxPath.Child("kind")
+			allErrs = append(allErrs, field.Invalid(kindPath, data.Kind, "the kind field of the extension status cannot be empty"))
+		}
+
+		if len(data.Type) == 0 {
+			typePath := idxPath.Child("type")
+			allErrs = append(allErrs, field.Invalid(typePath, data.Kind, "the type field of the extension status cannot be empty"))
+		}
+	}
+
+	return allErrs
+}
+
+// ValidateShootExtensionStatusSpecUpdate validates the update to the specification of a ShootExtensionStatus
+func ValidateShootExtensionStatusSpecUpdate(_, _ *core.ShootExtensionStatusSpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	return allErrs
+}
```

Comment:
Does it make sense to prevent kind/type from being changed?

Reply:
Type possibly yes, but not kind. Kind is the key in the list, so changing it is like adding a new entry and removing the old one.